CUSTOM serial and MIDI creation MACRO's

lathoub opened the pull request:
- MIDI_CREATE_CUSTOMSERIAL_INSTANCE: both overwritten serial and MIDI settings
- MIDI_CREATE_SPECIAL_INSTANCE: overwritten serial settings
(overwritten MIDI settings MACRO already exists)

lathoub committed:
CUSTOM serial and MIDI creation MACRO's

franky47 commented:
The number of different names for those macros is becoming quite high, it would be good to have some documentation about all the variants.

lathoub commented:
MACRO's are indeed dubieus (and not intuitive). I suggest to rethink the `MIDI_CREATE_*` macro naming **and** provide additional (wiki) documentation. Let me think about it over the weekend

franky47 commented:
Sounds good. We should keep the existing ones for compatibility in this version, since almost every single sketch out there uses those macros as the entrypoint.

lathoub commented:
Ideally `MIDI_CREATE_CUSTOM_INSTANCE` would be renamed to `MIDI_CREATE_CUSTOMMIDI_INSTANCE` , in line with  `MIDI_CREATE_CUSTOMSERIAL_INSTANCE`. `MIDI_CREATE_SPECIAL_INSTANCE` could then become `MIDI_CREATE_CUSTOMSERIALANDMIDI_INSTANCE`

Alternatively, only renamed `MIDI_CREATE_SPECIAL_INSTANCE` to `MIDI_CREATE_CUSTOMSERIALANDMIDI_INSTANCE` 

Either way: [update the wiki](https://github.com/FortySevenEffects/arduino_midi_library/wiki/Using-custom-Settings) as part of the PR/Release

lathoub commented:
Attached the updated wiki for the creation MACROs (not sure how to add this to this PR).

Should we go ahead with the renaming of the MACROs?

[Using-custom-Settings.md](https://github.com/FortySevenEffects/arduino_midi_library/files/10833453/Using-custom-Settings.md)

lathoub committed:
added examples for CUSTOM MACRO's

- added custom MIDI settings override example, disabling the 1byteparser
- added custom MIDI and Serial override example
- renamed CustomBaudRateSettings to CustomSerialSettings in existing CustomBaudRate example
- added DefaultSerialSettings to keywords
- added 2 examples to platformio.yml
